src/server/server_family.cc

```diff
@@ -978,6 +975,53 @@ void ServerFamily::Shutdown() {
   });
 }
 
+void ServerFamily::UpdateMemoryGlobalStats() {
+  ShardId sid = EngineShard::tlocal()->shard_id();
+  if (sid != 0) {  // This function is executed periodicaly on all shards. To ensure the logic
+                   // bellow runs only on one shard we return is the shard is not 0.
+    return;
+  }
+  time_t curr_time = time(nullptr);
+  if (curr_time == global_stats_update_time_) {  // Runs one a second.
```

Comment:
I see you changed semantics of the previous code by updating `used_mem_peak` once in a second. I am confused now whether we need to run `UpdateMemoryGlobalStats` as frequently as 1ms. Maybe you are right and it's not needed.  But I think it's an opportunity to simplify this logic so that timing rules will be clearer. 

Lets run UpdateMemoryGlobalStats once in 100ms, and the whole shard_handler will run once in 100ms. But then lets remove the 1s restriction here and update `used_mem_peak`, `rss_mem_current`, `rss_mem_peak` once in 100ms as wll

Reply:
ok I removed the logic to restrict this logic to run only once a second